velox/functions/sparksql/CMakeLists.txt

```diff
@@ -11,6 +11,14 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+# GCC 12 has a bug where it does not respect "pragma ignore" directives and ends
+# up failing compilation in an openssl header included by a hash-related
+# function.
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-declarations")
+endif()
```

Comment:
Have you tried  setting this via `target_compile_options(velox_functions_spark -Wno-deprecated-declarations)`?

Reply:
Not sure why I left the long comment on the last occurrence xD sorry

Reply:
Sorry I jumped the gun and got this merged before trying this out. I'll give it a try and if it works out, will submit a follow up PR to clean it up.